File: BoardGameTracker.Common/Models/PlayerStatistics.cs
namespace BoardGameTracker.Common.Models;

public class PlayerStatistics
{
    public int PlayCount { get; set; }
    public int WinCount { get; set; }
    public double TotalPlayedTime { get; set; }
    public int DistinctGameCount { get; set; }
    public List<MostPlayedGame> MostPlayedGames { get; set; } = [];
}

File: boardgametracker.client/src/models/Player/PlayerStatistics.ts
export interface PlayerStatistics {
  playCount: number;
  winCount: number;
  mostPlayedGames: MostPlayedGame[];
  totalPlayedTime: number;
  distinctGameCount: number;
}

export interface MostPlayedGame {
  id: number;
  image: string;
  title: string;
  totalWins: number;
  totalSessions: number;
  winningPercentage: number;
}
